R/mc_ls.R

```diff
@@ -25,3 +25,85 @@ mc_ls <- function(target, recursive = FALSE, flags = "") {
   cmd <- gsub("\\s+", " ", cmd)
   mc(cmd)
 }
+
+#' Directory listing as data frame
+#'
+#' This function parses the stdout from `mc_ls` and returns the result as a data frame.
+#'
+#' @param target Character string specifying the target directory path.
+#' @param recursive Logical indicating whether to recursively list directories.
+#'  Default is \code{FALSE}.
+#' @returns a data.frame with the directory listing information
+#' @export
+mc_ls_tbl <- function(target, recursive = FALSE) {
+  
+  con <- textConnection(encoding = "UTF-8", object = suppressMessages(
+    mc_ls(target=target, recursive=recursive)$stdout
+  ))
+  
+  on.exit(close(con))
+  
+  out <- readLines(con)
+  parse_mc_ls(trimws(out[nchar(out) > 0]))
+  
```

Comment:
Sorry if I'm missing it, but I the logic of the connection parsing seems a bit overkill here?  Modulo the empty last line, I believe that this is identical to the simpler approach of doing `strplit()` on the `stdout` text string?

Compare: 

```r
res <- mc("ls play/", verbose=FALSE)
con <- textConnection(encoding = "UTF-8", object = suppressMessages(
  res$stdout
))
con
out <- readLines(con)
out <- out[vapply(out, nchar, 1) > 0]

out2 <- strsplit(res$stdout, "\n")[[1]]

identical(out, out2)
```

also, note that `nchar(out)` isn't vectorized, so might not be doing what you want it to do here?

Reply:
Yes, thanks! I tried simplify accordingly.